File: invision-mobile/lib/src/core/enums/task_assignment_status.dart
import 'package:flutter/material.dart';

enum TaskAssignmentStatus {
  pending,
  inProgress,
  completed,
  verified,
  rejected;

  static TaskAssignmentStatus fromString(String value) {
    return switch (value) {
      'pending' => TaskAssignmentStatus.pending,
      'in_progress' => TaskAssignmentStatus.inProgress,
      'completed' => TaskAssignmentStatus.completed,
      'verified' => TaskAssignmentStatus.verified,
      'rejected' => TaskAssignmentStatus.rejected,
      _ => TaskAssignmentStatus.pending,
    };
  }

  String get value => switch (this) {
    TaskAssignmentStatus.pending => 'pending',
    TaskAssignmentStatus.inProgress => 'in_progress',
    TaskAssignmentStatus.completed => 'completed',
    TaskAssignmentStatus.verified => 'verified',
    TaskAssignmentStatus.rejected => 'rejected',
  };

  String get label => switch (this) {
    TaskAssignmentStatus.pending => 'Pending',
    TaskAssignmentStatus.inProgress => 'In Progress',
    TaskAssignmentStatus.completed => 'Completed',
    TaskAssignmentStatus.verified => 'Verified',
    TaskAssignmentStatus.rejected => 'Rejected',
  };

  Color get color => switch (this) {
    TaskAssignmentStatus.pending => Colors.orange,
    TaskAssignmentStatus.inProgress => Colors.blue,
    TaskAssignmentStatus.completed => Colors.green,
    TaskAssignmentStatus.verified => Colors.indigo,
    TaskAssignmentStatus.rejected => Colors.red,
  };
}

File: invision-backend/app/Enums/TaskAssignmentStatus.php
<?php

namespace App\Enums;

enum TaskAssignmentStatus: string
{
    case Pending = 'pending';
    case InProgress = 'in_progress';
    case Completed = 'completed';
    case Verified = 'verified';
    case Rejected = 'rejected';

    public function label(): string
    {
        return match ($this) {
            self::Pending => 'Pending',
            self::InProgress => 'In Progress',
            self::Completed => 'Completed',
            self::Verified => 'Verified',
            self::Rejected => 'Rejected',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Pending => 'yellow',
            self::InProgress => 'blue',
            self::Completed => 'green',
            self::Verified => 'indigo',
            self::Rejected => 'red',
        };
    }
}
